In [115]:
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns

Loading the CSV files containing our Youtube Trending Videos data.

In [116]:
ytdata = pd.read_csv('/Users/mac/Documents/Oasis_Infobyte_Project/YouYube Trending Videos data/GBvideos.csv')
pd.set_option('display.max_column',200)
ytdata.head()

video_id trending_date  \
0  Jw1Y-zhQURU      17.14.11   
1  3s1rvMFUweQ      17.14.11   
2  n1WpP7iowLc      17.14.11   
3  PUTEiSjKwJU      17.14.11   
4  rHwDegptbI4      17.14.11   

                                               title  \
0      John Lewis Christmas Ad 2017 - #MozTheMonster   
1          Taylor Swift: …Ready for It? (Live) - SNL   
2         Eminem - Walk On Water (Audio) ft. Beyoncé   
3  Goals from Salford City vs Class of 92 and Fri...   
4  Dashcam captures truck's near miss with child ...   

                channel_title  category_id              publish_time  \
0                  John Lewis           26  2017-11-10T07:38:29.000Z   
1         Saturday Night Live           24  2017-11-12T06:24:44.000Z   
2                  EminemVEVO           10  2017-11-10T17:00:03.000Z   
3  Salford City Football Club           17  2017-11-13T02:30:38.000Z   
4            Cute Girl Videos           25  2017-11-13T01:45:13.000Z   

                                                tags     views   likes  \
0  christmas|"john lewis christmas"|"john lewis"|...   7224515   55681   
1  SNL|"Saturday Night Live"|"SNL Season 43"|"Epi...   1053632   25561   
2  Eminem|"Walk"|"On"|"Water"|"Aftermath/Shady/In...  17158579  787420   
3  Salford City FC|"Salford City"|"Salford"|"Clas...     27833     193   
4                                             [none]      9815      30   

   dislikes  comment_count                                  thumbnail_link  \
0     10247           9479  https://i.ytimg.com/vi/Jw1Y-zhQURU/default.jpg   
1      2294           2757  https://i.ytimg.com/vi/3s1rvMFUweQ/default.jpg   
2     43420         125882  https://i.ytimg.com/vi/n1WpP7iowLc/default.jpg   
3        12             37  https://i.ytimg.com/vi/PUTEiSjKwJU/default.jpg   
4         2             30  https://i.ytimg.com/vi/rHwDegptbI4/default.jpg   

   comments_disabled  ratings_disabled  video_error_or_removed  \
0              False             False                   False   
1              False             False                   False   
2              False             False                   False   
3              False             False                   False   
4              False             False                   False   

                                         description  
0  Click here to continue the story and make your...  
1  Musical guest Taylor Swift performs …Ready for...  
2  Eminem's new track Walk on Water ft. Beyoncé i...  
3  Salford drew 4-4 against the Class of 92 and F...  
4  Dashcam captures truck's near miss with child ...

In [117]:
ytdata.dtypes

video_id                  object
trending_date             object
title                     object
channel_title             object
category_id                int64
publish_time              object
tags                      object
views                      int64
likes                      int64
dislikes                   int64
comment_count              int64
thumbnail_link            object
comments_disabled           bool
ratings_disabled            bool
video_error_or_removed      bool
description               object
dtype: object

In [118]:
ytdata.shape

(38916, 16)

Dealing with missing values:

In [119]:
ytdata.isna().sum()
#ytdata.loc[ytdata.description.isna()]

video_id                    0
trending_date               0
title                       0
channel_title               0
category_id                 0
publish_time                0
tags                        0
views                       0
likes                       0
dislikes                    0
comment_count               0
thumbnail_link              0
comments_disabled           0
ratings_disabled            0
video_error_or_removed      0
description               612
dtype: int64

- We can see from our output that only the description column has 'na' values (612).
- We will replace the null cells with 'N/A'.

In [120]:
ytdata.description.fillna('N/A', inplace=True)
ytdata.description.isna().sum()

/var/folders/97/31pwhl9978z102kvp11b4wzr0000gn/T/ipykernel_37892/1492923925.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  ytdata.description.fillna('N/A', inplace=True)


np.int64(0)

In [121]:
#Example:
ytdata.loc[[91]]

video_id trending_date              title  channel_title  category_id  \
91  _wM_jY_rass      17.14.11  Bone on Labour HQ  Ross Kempsell           25   

                publish_time    tags   views  likes  dislikes  comment_count  \
91  2017-11-06T16:12:58.000Z  [none]  159065   1160      1164              0   

                                    thumbnail_link  comments_disabled  \
91  https://i.ytimg.com/vi/_wM_jY_rass/default.jpg               True   

    ratings_disabled  video_error_or_removed description  
91             False                   False         N/A

In [122]:
ytdata.isnull().sum()

video_id                  0
trending_date             0
title                     0
channel_title             0
category_id               0
publish_time              0
tags                      0
views                     0
likes                     0
dislikes                  0
comment_count             0
thumbnail_link            0
comments_disabled         0
ratings_disabled          0
video_error_or_removed    0
description               0
dtype: int64

Removing Duplicates:

In [123]:
ytdata = ytdata.loc[~ytdata.duplicated()]

Data Standardization:

In [124]:
ytdata['trending_date']= pd.to_datetime(ytdata['trending_date'], format='%y.%d.%m')
ytdata.head(2)

video_id trending_date                                          title  \
0  Jw1Y-zhQURU    2017-11-14  John Lewis Christmas Ad 2017 - #MozTheMonster   
1  3s1rvMFUweQ    2017-11-14      Taylor Swift: …Ready for It? (Live) - SNL   

         channel_title  category_id              publish_time  \
0           John Lewis           26  2017-11-10T07:38:29.000Z   
1  Saturday Night Live           24  2017-11-12T06:24:44.000Z   

                                                tags    views  likes  \
0  christmas|"john lewis christmas"|"john lewis"|...  7224515  55681   
1  SNL|"Saturday Night Live"|"SNL Season 43"|"Epi...  1053632  25561   

   dislikes  comment_count                                  thumbnail_link  \
0     10247           9479  https://i.ytimg.com/vi/Jw1Y-zhQURU/default.jpg   
1      2294           2757  https://i.ytimg.com/vi/3s1rvMFUweQ/default.jpg   

   comments_disabled  ratings_disabled  video_error_or_removed  \
0              False             False                   False   
1              False             False                   False   

                                         description  
0  Click here to continue the story and make your...  
1  Musical guest Taylor Swift performs …Ready for...

In [125]:
#ytdata.columns
ytdata.rename(columns={'video_id':'Video_id','trending_date':'Trending_date','title':'Title',\
    'channel_title':'Channel_title','category_id':'Category_id','publish_time':'Publish_time', 'tags':'Tags', \
       'views':'Views','likes':'Likes', 'dislikes':'Dislikes','comment_count':'Comment_count',
       'thumbnail_link':'Thumbnail_link', 'comments_disabled':'Comments_disabled','ratings_disabled':'Ratings_disabled',
       'video_error_or_removed':'Video_error_or_removed', 'description':'Description'}, inplace=True) 

In [126]:
#ytdata['Category_id'].max()

In [127]:
#ytdata

Finding Outliers For 'Views,Likes,Dislikes,Comment_count' by 'Category_id':

In [128]:
ytdata[['Category_id','Views','Likes','Dislikes','Comment_count']].describe()

Category_id         Views         Likes      Dislikes  Comment_count
count  38745.000000  3.874500e+04  3.874500e+04  3.874500e+04   3.874500e+04
mean      16.825474  5.911293e+06  1.344673e+05  7.605730e+03   1.308039e+04
std        7.752308  1.901245e+07  3.498210e+05  5.104554e+04   5.069947e+04
min        1.000000  8.510000e+02  0.000000e+00  0.000000e+00   0.000000e+00
25%       10.000000  2.509390e+05  5.882000e+03  2.000000e+02   6.770000e+02
50%       20.000000  9.799490e+05  2.516800e+04  8.200000e+02   2.475000e+03
75%       24.000000  3.675563e+06  1.140220e+05  3.354000e+03   9.223000e+03
max       43.000000  4.245389e+08  5.613827e+06  1.944971e+06   1.626501e+06

In [129]:
Q1 = ytdata[['Category_id','Views','Likes','Dislikes','Comment_count']].describe().loc['25%']
Q3 = ytdata[['Category_id','Views','Likes','Dislikes','Comment_count']].describe().loc['75%']

In [130]:
IQR = Q3 - Q1 

In [131]:
lower_bound = Q1-1.5*IQR
upper_bound = Q3+1.5*IQR

In [132]:
print (lower_bound)

Category_id          -11.0
Views           -4885997.0
Likes            -156328.0
Dislikes           -4531.0
Comment_count     -12142.0
dtype: float64


In [133]:
print(upper_bound)

Category_id           45.0
Views            8812499.0
Likes             276232.0
Dislikes            8085.0
Comment_count      22042.0
dtype: float64


In [134]:
view_outlier = ytdata.query('Views > 8812499').shape[0] 
likes_outlier = ytdata.query('Likes > 276232').shape[0]
dislikes_outlier = ytdata.query('Dislikes > 8085').shape[0]
comment_count_outlier = ytdata.query('Comment_count > 22042').shape[0]
print ('No of Views Outliers:', view_outlier )
print ('No of likes Outliers:', likes_outlier )
print ('No of Dislikes Outliers:', dislikes_outlier )
print ('No of Comment count Outliers:', comment_count_outlier )

No of Views Outliers: 5295
No of likes Outliers: 4683
No of Dislikes Outliers: 5608
No of Comment count Outliers: 4900


Note:
- From our Analysis above We can deduce that there are no Lower Bound outliers.